tests/cipethernetip/cip_ethernet_ip_simulator.py

```
@@ -0,0 +1,59 @@
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import time
+from cpppo.server.enip import client
+from cpppo.server.enip.getattr import attribute_operations
+import json
+
+HOST = "127.0.0.1"
+
+with client.connector(host=HOST) as conn:
+        i =1 
+        while(1):
+        	try:
+                	data = [int(i) for c in range( 1 )]
+                        elements = len (data)
+		        tagType = client.enip.DINT.tag_type
+	     	        tag = "Scada"
+			req = conn.write( tag, elements=elements, data=data,
+                                   tag_type=tagType)
+
+          	except socket.error as exc:
```

Now what happens when other errors are received? Program stops? Should we have a catch all exceptions block and print the Exception?

---

I am not sure what has to be done here. Should I catch all exceptions?

---

I checked the client file in cpppo library, shutdown is not required, closing the connection is sufficient. so changed accordingly.